share/functions/prompt_pwd.fish

```diff
@@ -8,18 +8,34 @@ function prompt_pwd --description "Print the current working directory, shortene
         return 0
     end
 
-    # This allows overriding fish_prompt_pwd_dir_length from the outside (global or universal) without leaking it
+    # This allows overriding fish_prompt_pwd_dir_length and fish_prompt_pwd_depth from the outside (global or universal) without leaking it
     set -q fish_prompt_pwd_dir_length
     or set -l fish_prompt_pwd_dir_length 1
+  
+    set -q fish_prompt_pwd_depth
+    or set -l fish_prompt_pwd_depth 0
 
     # Replace $HOME with "~"
     set realhome ~
     set -l tmp (string replace -r '^'"$realhome"'($|/)' '~$1' $PWD)
+    # Split $temp to an array(for the trim depth only) 
+    set -l tempArr (string split '/' $tmp)
 
     if [ $fish_prompt_pwd_dir_length -eq 0 ]
-        echo $tmp
+
+        if [ $fish_prompt_pwd_depth -gt (count $tempArr) -o $fish_prompt_pwd_depth -eq 0 ] #dont trim, dont shorten
+            echo $tmp 
+        else #trim, dont shorten
+            echo (string join '/' $tempArr[(math "-1 * "$fish_prompt_pwd_depth"")..-1]) 
+        end
     else
-        # Shorten to at most $fish_prompt_pwd_dir_length characters per directory
-        string replace -ar '(\.?[^/]{'"$fish_prompt_pwd_dir_length"'})[^/]*/' '$1/' $tmp
+
+        if [ $fish_prompt_pwd_depth -gt (count $tempArr) -o $fish_prompt_pwd_depth -eq 0 ] #dont trim, shorten
```

Comment:
If you turn this condition around and just assign to $tmp directly, you can push the `string replace` out of this block and avoid duplicating it.

I.e.

```fish
# Not sure if it shouldn't be -lt - I'm assuming you don't want to remove the entire $PWD.
if [ $fish_prompt_pwd_depth -gt 0 -a $fish_prompt_pwd_depth -le (count $tempArr) ]
    set tmp (string join '/' # ...and so on
end
string replace -ar # ...the dir_length stuff
```